rosetta-source/src/main/rosetta/base-staticdata-asset-common-type.rosetta

```diff
@@ -95,13 +158,43 @@ type IndexReferenceInformation: <"A class defining information related to Index"
     condition IndexAttributes: <"A required choice condition for either a floating rate or inflation rate index.">
         indexName exists or indexId exists
 
-type Index extends ProductBase: <"Identifies an index by referencing a product identifier.">
+// ART-02 Observable and Index refactoring
```

Comment:
**PROPOSAL**
- whenever/anywhere an Observable type is used somewhere in the model, notably in Payout, as well as any other key payout argument such as Price or Quantity, then it shall be [**metadata address**], so that such Observable originally defined in TradeLot->PriceQuantity->Observable with [**metadata location**] can be referenced
- just rephrasing same idea : **booking method** below shall apply in generic manner accross whole model  : 
  - all Observable, Price, Quantity shall be defined in TradeLot->PriceQuantity with **metadata location**
  - then any usage of such type in Payout shall be referencing them with **metadata address**

**RATIONALE**
- as of today, model is **inconsistent** about "booking method" above described, beacuse it is partially in place for some Payout components (**_examples 1_**), and not in place for other ones (**_example 2_**)
- there are multiple **negative side-effects** for cases where referencing is not place, notably **DRR** e.g. see **_example 2_** below about ASIC new data fields for Barrier Price items 58 and 59
- existence of generic/consistent referencing from TradeLot items to Payout items is MILESTONE prerequisite for preparing **SmartContract** features (= input -> resolvingPayout -> output) which requires that input can be resolved in the place it is used, typically metadata location shall be leveraged for this purpose for producing resolved output in metadata addresse - hence the need to have this structure in place accross all Payout

**_Example 1_** - Referencing TradeLot->PriceQuantity in Payout terms exists
_non-exhaustive list_ 
- Payout->Underlier->[...]->product Identifier
- Payout->rateSpecification->rateOption
- Payout->rateSpecification->spreadSchedule
- Payout->ResolvablePriceQuantity->Quantity
- Payout->ResolvablePriceQuantity->Price (by the way, highly recommand to remove this one, in order to ensure that any Price component is an attribute of core payout, in order for CDM to be aligned ith generic payout schema that is Quantity x CorePayout... in regards of which CDM situation is confusing, because we have PriceQuantity x Payout, where some Price might be out of Payout per se and Payout may not include any core formula component at all, for instance the weird case of fixedPricePayout, which implicit functional formula would be = Quantity x Price hence no Underlier neither Fixed Price in core payout exist...)

**_Example 2_** - Referencing TradeLot->PriceQuantity in Payout terms is **missing**
_non-exhaustive list_ 
- trigger and triggerPercentage are of type "number" in Trigger type : this one is causing issue for DRR
- optionStrike is Price but with no metadata
- volatilityStrike, same comment
- varianceStrike, same comment
- valuationPriceInitial, same comment